packages/backend/src/services/RenameService/rename.ts

```diff
@@ -628,13 +636,15 @@ export const renameSavedChart = ({
     validate: boolean;
 }): { updatedChart: SavedChartDAO; hasChanges: boolean } => {
     let hasChanges = false;
-
     const isPrefix = type === RenameType.MODEL;
-    const containsModelName = (object: Object) =>
-        JSON.stringify(object).includes(
-            `${nameChanges.from}${isPrefix ? '_' : ''}`,
-        );
 
+    const searchTerms = Object.values({
+        from: addSuffixIfPrefix(nameChanges.from, isPrefix),
+        fromReference: addSuffixIfPrefix(nameChanges.fromReference, isPrefix),
+        fromFieldName: addSuffixIfPrefix(nameChanges.fromFieldName, isPrefix),
```

Comment:
I understand looking at fromReference, but why fieldName ? if we don't need it, I'd suggest to remove it to avoid false positives.